### PR TITLE
Adds git-repo-scanner

### PR DESCRIPTION
**Pull request by paulschmelzer**
Adds a scanner to discover git repositories on github or gitlab. The main purpose of this scanner is running cascades on gitleaks.

### CODE REVIEW

**Comment by J12934** [comment]
Awesome 👍
Will take a deeper look later today / tomorrow

File: scanners/git-repo-scanner/parser/__testFiles__/git-scanner-test-findings.json [hunk 1]
```diff
@@ -0,0 +1,542 @@
+[
+  {
+    "name": "Github Repo",
+    "description": "A Github repository",
+    "category": "Git Repository",
+    "osi_layer": "APPLICATION",
+    "severity": "INFORMATIONAL",
+    "attributes": {
+      "id": 80711933,
+      "web_url": "https://api.github.com/repos/secureCodeBox/secureCodeBox",
```
Comment (J12934) [inline scanners/git-repo-scanner/parser/__testFiles__/git-scanner-test-findings.json:10]: Is this really correct with api.github.com?
I'd assume that gitleaks could not handle this.

File: scanners/git-repo-scanner/helm2.Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+name: git-repo-scanner
+description: A Helm chart for the git-repo-scanner that integrates with the secureCodeBox.
+
+type: application
+# version - gets automatically set to the secureCodeBox release version when the helm charts gets published
+version: latest
+appVersion: v0.1
```
Comment (J12934) [inline scanners/git-repo-scanner/helm2.Chart.yaml:8]:
```suggestion
```

File: scanners/git-repo-scanner/values.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+image:
+  # image.repository -- Container Image to run the scan
+  repository: docker.io/securecodebox/parser-git-repo-scanner
+  # image.tag -- defaults to the charts version
+  tag: null
+
+parserImage:
+  # parserImage.tag - defaults to the charts version
+  # parserImage.repository -- Parser image repository
+  repository: docker.io/securecodebox/scanner-git-repo-scanner
```
Comment (J12934) [inline scanners/git-repo-scanner/values.yaml:10]:
```suggestion
  repository: docker.io/securecodebox/parser-git-repo-scanner
```

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -319,6 +329,14 @@ jobs:
...
           path: ./scanners/gitleaks/scanner/
           tags: "v6.1.2,latest"
+      - uses: docker/build-push-action@v1
+        name: "Build & Push Git-Repo-Scanner Image"
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+          repository: securecodebox/scanner-git-repo-scanner
+          path: ./scanners/git-repo-scanner/scanner/
+          tags: "v0.1,latest"
```
Comment (J12934) [inline .github/workflows/ci.yaml:339]: See versioning notes in Chart.yaml

```suggestion
          tags: "latest"
          tag_with_ref: true
```

**Review by J12934 — CHANGES_REQUESTED** [review]

File: scanners/git-repo-scanner/cascading-rules/gitleaks-scan.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+kind: CascadingRule
+metadata:
+  name: "gitleaks-scan"
+  labels:
+    securecodebox.io/invasive: non-invasive
+    securecodebox.io/intensive: high
+spec:
+  matches:
+    anyOf:
+      - category: "Git Repository"
```
[the remark this answers is not in the document]
Comment (paulschmelzer) [reply scanners/git-repo-scanner/cascading-rules/gitleaks-scan.yaml:11]: I changed the cascade rules and added one.

File: scanners/git-repo-scanner/parser/__testFiles__/git-scanner-test-findings.json (hunk 1 above)
Comment (paulschmelzer) [reply scanners/git-repo-scanner/parser/__testFiles__/git-scanner-test-findings.json:10]: Yes!! Thank you 👍

**Review by J12934 — APPROVED** [review]

**Pull request merged**